rotkehlchen/chain/manager.py

```diff
@@ -621,10 +621,18 @@ def query_balances(
             self.query_polkadot_balances(ignore_cache=ignore_cache)
         if should_query_avax:
             self.query_avax_balances(ignore_cache=ignore_cache)
+        if ignore_cache is True:
+            self.derive_new_addresses_from_xpubs()
```

Comment:
Shouldn't this happen before btc and bch balances are queried? Also this should happen only if BTC or BCH are queried

Reply:
The order in which they're queried shouldn't change anything as the totals are eventually recalculated. I might be wrong though.

Yeah, you're right. It should only be done if btc/bch are queried.